### PR TITLE
initial port of V8 and its deps

### PR DESCRIPTION
**Pull request by gabylb**
The port in this PR applies as of Nov 22 commit [fce38915e4c7c73c0e7f77bb13e2a78514968d35](https://chromium.googlesource.com/v8/v8/+/fce38915e4c7c73c0e7f77bb13e2a78514968d35). See README.md for details on how to apply the patches.

Current test results from release build (`+pass|-fail`):
- cctest `+4607|-509`
- cmkgrokdump `+0|-0`
- debugger `+312|-0`
- fuzzer `+0|-55`
- intl `+299|-0`
- inspector `+382|-7`
- message `+363|-0`
- mjsunit `+5990|-9`
- unittests `+5268|-19`
- wasm-api-tests `+7|-12`
- wasm-js `+103|-9`
- wasm-spec-tests `+337|-0`

### CODE REVIEW

File: patches/build/git.20231122.968682938b.diff [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
++  cxx = "clang++"
++  asm = "clang"
+   ar = "ar"
+   ld = cxx
+ 
+@@ -36,7 +36,7 @@ toolchain("s390x") {
+ 
+   tool("cc") {
+     depfile = "{{output}}.d"
+-    command = "$cc -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_c}}${extra_cflags} -c {{source}} -o {{output}}"
```
Comment (MikeFultonDev) [inline patches/build/git.20231122.968682938b.diff:120]: will this break Linux on Z?

File: patches/build/git.20231122.968682938b.diff [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+-    command = "$asm -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
++    command = "$asm ${rebuild_string} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
+     depsformat = "gcc"
+     description = "ASM {{output}}"
+     outputs = [ "$object_subdir/{{source_name_part}}.o" ]
+@@ -88,7 +88,7 @@ toolchain("s390x") {
+     # implement (1) linking to produce a .so, (2) extracting the symbols
+     # from that file (3) if the extracted list differs from the existing
+     # .TOC file, overwrite it, otherwise, don't change it.
+-    link_command = "$ld -Wl,DLL {{ldflags}}${extra_ldflags} -o \"$sofile\" `cat $rspfile`"
```
Comment (MikeFultonDev) [inline patches/build/git.20231122.968682938b.diff:147]: will this break Linux on Z?

**Review by MikeFultonDev — APPROVED** [review]
couple minor comments - LGTM

File: patches/build/git.20231122.968682938b.diff (hunk 2 above)
Comment (netskink) [reply patches/build/git.20231122.968682938b.diff:147]: I have no clue. I’m on holiday.  Do we wish to merge today or wait until Monday?

File: patches/build/git.20231122.968682938b.diff (hunk 1 above)
Comment (gabylb) [reply patches/build/git.20231122.968682938b.diff:120]: No, this is a z/OS-only file `toolchain/zos/BUILD.gn`.

Comment (gabylb) [reply patches/build/git.20231122.968682938b.diff:120]: No, this file is a z/OS-only file `toolchain/zos/BUILD.gn` (line 99).

File: patches/build/git.20231122.968682938b.diff (hunk 2 above)
Comment (gabylb) [reply patches/build/git.20231122.968682938b.diff:147]: No, this file is a z/OS-only file `toolchain/zos/BUILD.gn` (line 99).

Comment (gabylb) [reply patches/build/git.20231122.968682938b.diff:147]: @netskink I'm hoping Igor will have a chance to review it first before merging.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Comment by netskink** [comment]
Thanks! I appreciate your work Gaby!

**Pull request merged**